<launch>

  <arg name="manager" value="vzense_manager"/>
  <arg name="depth_threshold" default="1.5"/>
  <arg name="bagpath" />
  <arg name="camera" default="camera" />
  <arg name="republish_ns" default="remote" />
  <arg name="queue_size" default="30" />

  <group ns="$(arg republish_ns)">
    <group ns="left_vzense_camera" >

      <node name="$(arg manager)"
            pkg="nodelet" type="nodelet"
            args="manager"/>

      <group ns="color" >
        <node name="republish"
              pkg="image_transport" type="republish"
              args="compressed raw">
          <remap from="in" to="/$(arg camera)/left_vzense_camera/color/image_raw" />
          <remap from="out" to="image_raw" />
        </node>
      </group>

      <group ns="transformedDepth" >
        <node name="republish"
              pkg="image_transport" type="republish"
              args="compressedDepth raw">
          <remap from="in" to="/$(arg camera)/left_vzense_camera/transformedDepth/image_raw" />
          <remap from="out" to="image_raw" />
        </node>
      </group>

      <node name="depth_image_filter"
            pkg="vzense_demo" type="depth_image_filter.py"
            output="screen" respawn="true">
        <remap from="~input" to="transformedDepth/image_raw" />
        <remap from="~output" to="transformedDepth/filtered/image_raw" />
        <remap from="~output/mask" to="transformedDepth/image_raw/mask" />
        <rosparam subst_value="true" >
          threshold: $(arg depth_threshold)
        </rosparam>
      </node>

      <node pkg="nodelet" type="nodelet" name="pointcloudxyzrgb_manager"
            args="load depth_image_proc/point_cloud_xyzrgb $(arg manager)"
            clear_params="true" >
        <remap from="depth_registered/image_rect" to="transformedDepth/filtered/image_raw"/>
        <remap from="depth_registered/points" to="transformedDepth/points"/>
        <remap from="rgb/image_rect_color" to="color/image_raw"/>
        <remap from="rgb/camera_info" to="/$(arg camera)/left_vzense_camera/color/camera_info"/>
        <rosparam subst_value="true" >
          queue_size: $(arg queue_size)
        </rosparam>
      </node>

    </group>

  <!--   <group ns="right_vzense_camera" > -->

  <!--     <node name="$(arg manager)" -->
  <!--           pkg="nodelet" type="nodelet" -->
  <!--           args="manager"/> -->

  <!--     <group ns="color" > -->
  <!--       <node name="republish" -->
  <!--             pkg="image_transport" type="republish" -->
  <!--             args="compressed raw"> -->
  <!--         <remap from="in" to="image_raw" /> -->
  <!--         <remap from="out" to="image_raw" /> -->
  <!--       </node> -->
  <!--     </group> -->

  <!--     <group ns="transformedDepth" > -->
  <!--       <node name="republish" -->
  <!--             pkg="image_transport" type="republish" -->
  <!--             args="compressedDepth raw"> -->
  <!--         <remap from="in" to="image_raw" /> -->
  <!--         <remap from="out" to="image_raw" /> -->
  <!--       </node> -->
  <!--     </group> -->

  <!--     <node name="register" -->
  <!--           pkg="nodelet" type="nodelet" -->
  <!--           args="load depth_image_proc/register $(arg manager)" -->
  <!--           clear_params="true"> -->
  <!--       <remap from="rgb/camera_info"   to="color/camera_info"/> -->
  <!--       <remap from="depth/camera_info" to="depth/camera_info"/> -->
  <!--       <remap from="depth/image_rect"  to="depth/image_raw"/> -->
  <!--       <remap from="depth_registered/image_rect"  to="transformedDepth/image_raw"/> -->
  <!--       <remap from="depth_registered/camera_info"  to="transformedDepth/camera_info"/> -->
  <!--     </node> -->

  <!--     <node name="depth_image_filter" -->
  <!--           pkg="vzense_demo" type="depth_image_filter.py" -->
  <!--           output="screen" respawn="true"> -->
  <!--       <remap from="~input" to="transformedDepth/image_raw" /> -->
  <!--       <remap from="~output" to="transformedDepth/filtered/image_raw" /> -->
  <!--       <remap from="~output/mask" to="transformedDepth/image_raw/mask" /> -->
  <!--       <rosparam subst_value="true" > -->
  <!--         threshold: $(arg depth_threshold) -->
  <!--       </rosparam> -->
  <!--     </node> -->

  <!--     <node pkg="nodelet" type="nodelet" name="pointcloudxyzrgb_manager" -->
  <!--           args="load depth_image_proc/point_cloud_xyzrgb $(arg manager)" -->
  <!--           clear_params="true" > -->
  <!--       <remap from="depth_registered/image_rect" to="transformedDepth/filtered/image_raw"/> -->
  <!--       <remap from="depth_registered/points" to="transformedDepth/points"/> -->
  <!--       <remap from="rgb/image_rect_color" to="color/image_raw"/> -->
  <!--       <remap from="rgb/camera_info" to="color/camera_info"/> -->
  <!--     </node> -->

  <!--   </group> -->
  </group>

</launch>
